components/kyma-environment-broker/internal/process/operation_manager.go

```diff
@@ -29,7 +28,15 @@ func (om *OperationManager) OperationSucceeded(operation internal.Operation, des
 
 // OperationFailed marks the operation as failed and returns status of the operation's update
 func (om *OperationManager) OperationFailed(operation internal.Operation, description string, err error, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
-	return om.update(operation, domain.Failed, description, log)
+	op, t, updateErr := om.update(operation, domain.Failed, description, log)
```

Comment:
if t !=0, it should directly return `return op, t, nil`, right?

Reply:
I'm not sure it makes any difference. It is actually written twice, once by the step and it looks like on error, the new staged manager tries to write it to the db as well

https://github.com/kyma-project/control-plane/blob/b10c665c95baa1ab2d181747d65a4c50f2239c30/components/kyma-environment-broker/internal/process/staged_manager.go#L194-L205

but most importantly, when operation is marked as failed, the staged manager ignores the requeue period
https://github.com/kyma-project/control-plane/blob/b10c665c95baa1ab2d181747d65a4c50f2239c30/components/kyma-environment-broker/internal/process/staged_manager.go#L140-L148

Reply:
This is outside of m.runStep() already. inside m.runStep(), if less than 10*time.Minute, it will be retried.
https://github.com/kyma-project/control-plane/blob/0e8ca0d46165094a29087d4765d2477d8357b966/components/kyma-environment-broker/internal/process/staged_manager.go#L221

Reply:
ah, you are right!